source/common/http/conn_manager_utility.cc

```diff
@@ -47,6 +47,15 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
   request_headers.removeProxyConnection();
   request_headers.removeTransferEncoding();
 
+  // When this flag is true, we represent IPv4 addresses as IPv4 mapped IPv6 if
+  //   - use_remote_address is true &&
+  //   - the address was not derived from the existing XFF header &&
+  //   - the address is not a loopback address
+  const bool represent_ipv4_as_ipv6 = runtime.snapshot().featureEnabled(
+      "http_connection_manager.represent_ipv4_remote_address_as_ipv4_mapped_ipv6",
+      config.representIpv4RemoteAddressAsIpv4MappedIpv6() ? 100 : 0);
+  bool should_map_final_remote_address = false;
```

Comment:
From looking at the code, I'm getting confused about how `represent_ipv4_as_ipv6` relates to `should_map_final_remote_address`. Can you add more comments? Is there any way to simplify the logic further or is this as good as it gets?

Reply:
i'll add more comments, the gist of `should_map_final_remote_address` was to defer mapping until the end to avoid affecting some of the other logic